catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/fragment/CopyCompositeBrickTest.kt

```diff
@@ -0,0 +1,266 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.ui.fragment
+
+import androidx.test.espresso.DataInteraction
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.Espresso.openContextualActionModeOverflowMenu
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isChecked
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.isEnabled
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import org.catrobat.catroid.ProjectManager
+import org.catrobat.catroid.R
+import org.catrobat.catroid.content.Script
+import org.catrobat.catroid.content.bricks.ForeverBrick
+import org.catrobat.catroid.content.bricks.IfLogicBeginBrick
+import org.catrobat.catroid.content.bricks.SetXBrick
+import org.catrobat.catroid.content.bricks.SetYBrick
+import org.catrobat.catroid.ui.SpriteActivity
+import org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition
+import org.catrobat.catroid.uiespresso.formulaeditor.FormulaEditorMultiplayerVariablesTest
+import org.catrobat.catroid.uiespresso.util.UiTestUtils.Companion.createProjectAndGetStartScript
+import org.catrobat.catroid.uiespresso.util.rules.FragmentActivityTestRule
+import org.hamcrest.Matchers.allOf
+import org.hamcrest.Matchers.not
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.koin.java.KoinJavaComponent.inject
+
+class CopyCompositeBrickTest {
+
+    val projectManager: ProjectManager by inject(ProjectManager::class.java)
+
+    var firstIndexIfComposite = 1
+    var elseIndexIfComposite = 3
+    var lastIndexIfComposite = 5
+
+    var firstIndexForeverBrick = 6
+    var lastIndexForeverBrick = 10
+
+    @JvmField
+    @Rule
+    var baseActivityTestRule = FragmentActivityTestRule(
+        SpriteActivity::class.java,
+        SpriteActivity.EXTRA_FRAGMENT_POSITION,
+        SpriteActivity.FRAGMENT_SCRIPTS
+    )
+
+    @Before
+    @Throws(Exception::class)
+    fun setUp() {
+        val script = createProjectAndGetStartScript(
+            FormulaEditorMultiplayerVariablesTest::class.java.simpleName
+        )
+        addTestBricks(script)
+        baseActivityTestRule.launchActivity()
+    }
+
+    @Test
+    fun testEnabledInnerBricksOfCompositeBricks() {
+        openContextualActionModeOverflowMenu()
+        onView(withText(R.string.copy)).perform(click())
+
+        getCheckbox(firstIndexIfComposite)?.perform(click())
+
+        getCheckbox(firstIndexIfComposite + 1)
+            ?.check(matches(isEnabled()))
```

Comment:
use Ctrl+Alt+L (twice) to run autoformatter, this will remove your custom redundant linebreaks